front/src/modules/ui/table/states/viewFieldsState.ts

```diff
@@ -1,8 +1,49 @@
-import { atom } from 'recoil';
+import { atom, selector } from 'recoil';
 
-import { ViewFieldDefinition, ViewFieldMetadata } from '../types/ViewField';
+import { companyViewFields } from '@/companies/constants/companyViewFields';
+import { peopleViewFields } from '@/people/constants/peopleViewFields';
 
-export const viewFieldsState = atom<ViewFieldDefinition<ViewFieldMetadata>[]>({
+import type {
+  ViewFieldDefinition,
+  ViewFieldMetadata,
+} from '../types/ViewField';
+
+export const viewFieldsState = atom<{
+  objectName: 'company' | 'person' | '';
```

Comment:
I'm surprised about the '' possibility? Can we get rid of it?

Reply:
I added an empty possibility because I was not sure what value to use by default: `company` or `person`? Not sure if it makes sense to choose one when no `viewFields` have been fetched yet.

Reply:
Are forced to pass a default? Maybe it doesn't make sense for viewFields to have a default and it should always be set with something that makes sense